src/render/mod.rs

```diff
@@ -434,19 +466,29 @@ impl Renderer {
         let quad_buf = gl_factory.create_constant_buffer(1);
         let light_buf = gl_factory.create_constant_buffer(MAX_LIGHTS);
         let pbr_buf = gl_factory.create_constant_buffer(1);
+        let inst_buf = gl_factory
+            .create_buffer(
+                INSTANCE_COUNT,
```

Comment:
instead of keeping around a fixed-inst-count buffer, I think we should have an automatically grown instance buffer per shader. We collect CPU instances first, then see if the buffer needs to be re-allocated (and do this), then fill out instance data.

Reply:
If so, we need to store factory inside Renderer. Can't remember whether it's possible (probably yes)